src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -1419,28 +1363,7 @@ public override int ReadByte()
 
 #if USE_OVERLAPPED
         [System.Security.SecuritySafeCritical]  // auto-generated
-        internal IAsyncResult BeginWrite(byte[] array, int offset, int numBytes, AsyncCallback userCallback, Object stateObject)
-        {
-            if (array == null)
-                throw new ArgumentNullException("array");
-            if (offset < 0)
-                throw new ArgumentOutOfRangeException("offset", SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (numBytes < 0)
-                throw new ArgumentOutOfRangeException("numBytes", SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (array.Length - offset < numBytes)
-                throw new ArgumentException(SR.Argument_InvalidOffLen);
-            Contract.EndContractBlock();
-
-            if (_handle.IsClosed) throw __Error.GetFileNotOpen();
-
-            if (!_isAsync)
-                return _streamAsyncHelper.BeginWrite(array, offset, numBytes, userCallback, stateObject);
-            else
-                return BeginWriteAsync(array, offset, numBytes, userCallback, stateObject);
-        }
-
-        [System.Security.SecuritySafeCritical]  // auto-generated
-        private FileStreamAsyncResult BeginWriteAsync(byte[] array, int offset, int numBytes, AsyncCallback userCallback, Object stateObject)
+        private Task<int> BeginWriteAsync(byte[] array, int offset, int numBytes, CancellationToken cancellationToken)
```

Comment:
Why return a ```Task<int>``` instead of just a ```Task```?  Write doesn't need to return how many bytes were written, and in returning just the non-generic Task, you can more aggressively return a cached Task instance, e.g. Task.CompletedTask.  I don't remember if we're able to use Task.CompletedTask here yet, i.e. if the latest contract that CompletedTask is in is available to FileSystem, but if it's not, you can always define your own in a readonly static.